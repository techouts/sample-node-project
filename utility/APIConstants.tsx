const NEXT_PUBLIC_JUSPAY_APIURL = process.env.NEXT_PUBLIC_JUSPAY_APIURL;
export const merchant_id = "ssbeauty";
export const oauth_token = "authorizationserver/oauth/token";
export const otp_services = "services/v2_1/ssl/sendOTP/OB";
export const otp_service_verifier =
  "services/v2_1/ssl/users/loginOrSignup?mobileNumber=9640558040&otp=782253&cliend_id=mobile_android";
export const SignUp_client_id = "mobile_android";
export const getPaymentMethods = `${NEXT_PUBLIC_JUSPAY_APIURL}/merchants/${merchant_id}/paymentmethods?options.add_emandate_payment_methods=false&options.check_wallet_direct_debit_support=false&options.add_supported_reference_ids=false`;
export const CardBin = `${NEXT_PUBLIC_JUSPAY_APIURL}/cardbins/`;
export const CardTokenization = `${NEXT_PUBLIC_JUSPAY_APIURL}/card/tokenize`;
export const CardTransaction = `${NEXT_PUBLIC_JUSPAY_APIURL}/txns`;
export const QyPaymentStatus = `${NEXT_PUBLIC_JUSPAY_APIURL}/order/payment-status?merchant_id=ssbeauty&order_id=`;
export const paymentStatus =
  "/order/payment-status?merchant_id=ssbeauty&order_id=";
export const GOOGLE_MAP_DIRECTIONS_API = (
  origin: string,
  destination: string
) =>
  `${process.env.NEXT_PUBLIC_GOOGLE_MAP_URL}?api=1&origin=${origin}&destination=${destination}&travelmode=car`;
  export const UNBXD_HOST = process.env.NEXT_UNBXD_HOST;

  export const IS_UNBXD_ENABLED = process.env.NEXT_PUBLIC_IS_UNBXD_ENABLED === "true";

  export const MIDDLEWARE_UNBXD = process.env.NEXT_PUBLIC_MIDDLEWARE_UNBXD_SEARCH;

  export const IS_UNBXD_MIDDLEWARE_ENABLED = process.env.NEXT_PUBLIC_IS_UNBXD_MIDDLEWARE_ENABLED === "true"

  const NEXT_PUBLIC_D365_APP_ID_GUEST = process.env.NEXT_PUBLIC_D365_APP_ID_GUEST;

  const NEXT_PUBLIC_D365_APP_ID_LOGGED_IN = process.env.NEXT_PUBLIC_D365_APP_ID_LOGGED_IN;

  const NEXT_PUBLIC_D365_ORG_ID = process.env.NEXT_PUBLIC_D365_ORG_ID;
  
  const NEXT_PUBLIC_D365_ORG_URL = process.env.NEXT_PUBLIC_D365_ORG_URL;
  export const Google_API_KEY = "AIzaSyCBo5PdGTw5Z0BS6hTboacbxkMC7vIZ6FY";

  export {NEXT_PUBLIC_D365_APP_ID_LOGGED_IN, NEXT_PUBLIC_D365_APP_ID_GUEST, NEXT_PUBLIC_D365_ORG_ID, NEXT_PUBLIC_D365_ORG_URL}
  export const IS_RECAPTCHA_ENABLED = process.env.NEXT_PUBLIC_IS_RECAPTCHA_ENABLED === "true";