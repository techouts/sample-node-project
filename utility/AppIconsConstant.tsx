import { useRecoilState } from "recoil";
import { SSBLogos } from "../recoilstore";

export const AppIcons = (image: string) => {
  const [SSBeautyLogos] = useRecoilState(SSBLogos);

  const imageIcon = SSBeautyLogos?.appLogos?.find(
    (item: any) => item?.name == image
  );
  return imageIcon;
};
