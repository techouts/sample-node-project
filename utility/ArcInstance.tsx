import axios from "axios";
import { Cookies } from "react-cookie";

const cookie = new Cookies();
const NEXT_PUBLIC_ORC_LAYER = process.env.NEXT_PUBLIC_ORC_LAYER;

export const arcClient = axios.create({
  baseURL: NEXT_PUBLIC_ORC_LAYER,
});

arcClient.interceptors.request.use((config) => {
  const accessToken = `${cookie.get("accessToken")}`;
  const axiosConfig = { ...config };
  axiosConfig.headers["Content-Type"] = "application/json";
  axiosConfig.headers["Access-Control-Allow-Origin"] = "*";
  axiosConfig.headers["Accept"] = "*/*";
  axiosConfig.headers["Accept-Encoding"] = "gzip, deflate, br";
  axiosConfig.headers["Connection"] = "keep-alive";
  if (accessToken && accessToken !== "undefined") {
    axiosConfig.headers.Authorization = `Bearer ${accessToken}`;
  }
  return axiosConfig;
});