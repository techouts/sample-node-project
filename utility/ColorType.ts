import { type } from "os";

type RGB = `rgb(${number}, ${number}, ${number})`;
type RGBA = `rgba(${number}, ${number}, ${number}, ${number})`;
type HEX = `#${string}`;

export type ColorType = RGB | RGBA | HEX;