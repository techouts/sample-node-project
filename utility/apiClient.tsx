import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
import { callErrorEvent } from "./ErrorEventAnalytics";
import { fetchMiddleWareAccessToken } from "./MiddelwareAccessWrapper";

const GRAPHQL_URL = process.env.NEXT_PUBLIC_GRAPHQL_URL!;
const MIDDLEWARE_URL = process.env.NEXT_PUBLIC_SSO_API_GATEWAY_URL!;
const isMiddlewareSecurityEnabled =
  process.env.NEXT_PUBLIC_IS_MIDDLEWARE_SECURITY_ENABLED === "true";

const axiosInstance = axios.create();

async (config: AxiosRequestConfig<any>) => {
  if (!config.url?.includes("/middleware")) return config;

  const token = localStorage.getItem("accessToken");
  const updatedHeaders: AxiosRequestConfig["headers"] = {
    ...config.headers,
    authorization: token ? `Bearer ${token}` : "",
  };
  if (isMiddlewareSecurityEnabled) {
    let mToken = localStorage.getItem("m-token");
    if (mToken && mToken !== "") {
      updatedHeaders["x-auth-token"] = `Bearer ${mToken}`;
    } else {
      const mTokenRes = await fetchMiddleWareAccessToken();
      if (mTokenRes?.data?.access_token) {
        updatedHeaders[
          "x-auth-token"
        ] = `Bearer ${mTokenRes?.data?.access_token}`;
      }
    }
  }

  return { ...config, headers: updatedHeaders };
};

axiosInstance.interceptors.response.use(
  (response: AxiosResponse) => {
    return response;
  },
  async (error: any) => {
    if (error.response) {
      const { message, status } = error.response.data || {};
      console.error(`[Error]: ${message}, Status: ${status}`);

      let params = {
        error_message: message || error.message,
        status: status || error.response?.status,
        item_name: GRAPHQL_URL,
      };

      callErrorEvent(params);
    } else if (error.request) {
      console.error("[Network error]: No response received");
    } else {
      console.error(`[Error]: ${error.message}`);
    }
    return Promise.reject(error);
  }
);

// Retry logic for requests
const retryRequest = async (
  config: AxiosRequestConfig<any>,
  retries = 3
): Promise<AxiosResponse | void> => {
  let lastError;

  for (let i = 0; i < retries; i++) {
    try {
      return await axiosInstance(config);
    } catch (error: any) {
      lastError = error;

      if (error.response?.status === 401 || error.response?.status === 403) {
        console.warn(
          `Attempt ${i + 1}: Received ${
            error.response.status
          }, trying to refresh token...`
        );
        const mTokenRes = await fetchMiddleWareAccessToken();
        const mToken = mTokenRes?.data?.access_token;

        if (!mToken) {
          console.error("Token refresh failed. Aborting retries.");
          break;
        }
        localStorage.setItem("m-token", mToken);
        config = {
          ...config,
          headers: {
            ...(config.headers || {}),
            "x-auth-token": `Bearer ${mToken}`,
          },
        };

        continue;
      }

      console.error(
        `Request failed with status ${error.response?.status}. Retrying...`
      );
    }
  }

  return Promise.reject(lastError);
};

// Function to send requests
const sendRequest = async (
  query: string,
  isCMSQuery = false,
  middleWareURL = false,
  urlKey?: string
): Promise<AxiosResponse | void> => {
  const url = urlKey
    ? `${urlKey}?query=${query}`
    : isCMSQuery
    ? query
    : middleWareURL
    ? `${MIDDLEWARE_URL}?query=${query}`
    : `${GRAPHQL_URL}?query=${query}`;

  const config: AxiosRequestConfig = {
    method: "get",
    url: url,
    headers: {},
  };

  try {
    return await retryRequest(config);
  } catch (error: any) {
    console.error(`Request failed: ${error}`);
    return Promise.reject(error);
  }
};

export default sendRequest;
