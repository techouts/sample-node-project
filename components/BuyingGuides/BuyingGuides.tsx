import Box from "@mui/material/Box";
import Stack from "@mui/material/Stack";
import {
  BuyingGuideInterface,
  ITEMS_LIST,
} from "../../schemas/BuyingGuidesSchema";
import { ReplaceImage } from "../../utility/ReplaceImage";
import {
  Styledimage,
  SubTitleTypography,
  ViewAllbutton,
  MainBox,
  StyledGrid,
  TitleTypography,
  SubTitle,
  StyledBox,
  BoxStyled,
} from "./Styles";
import { useRouter } from "next/router";
import { useMobileCheck } from "../../utility/isMobile";
import Carousel from "../../HOC/Carousel/Carousel2";
import Title from "../../HOC/Title/Title";
import ViewEvent from "../../utility/viewEvent";
import { widget_type, event_type } from "../../utility/GAConstants";
import { useEffect, useRef, useState } from "react";
import triggerGAEvent from "../../utility/GaEvents";
import { viewArray } from "../../utility/ViewEvenItemArray";
import TriggerGaViewPromotion from "../../utility/GAViewPromotion";
import { Cookies } from "react-cookie";
import useStorage from "../../utility/useStoarge";
import { useInView } from "react-intersection-observer";

export const MsiteCarousel = ({
  imageUrl,
  text,
  mobileTitle,
  isNewtab,
  path,
  textColor,
  promotion_id,
  promotion_name,
  creative_name,
  creative_slot,
}: ITEMS_LIST) => {
  const router = useRouter();

  const cookie = new Cookies();
  const { getItem } = useStorage();

  const callSelectPromotionEvent = () => {
    const datasLayer = {
      previous_screen_name: "na",
      visitor_type: cookie.get("accessToken") ? "loggedin" : "Guest",
      user_id: getItem("customer_ref", "local") || "na",
      creative_name: `${creative_name}-outer BGDM`,
      creative_slot: `${creative_slot}-outer BGDM`,
      items: [
        {
          promotion_id: `${promotion_id}-inner BGDM`,
          promotion_name: `${promotion_name}-inner BGDM`,
          creative_name: `${creative_name}-inner BGDM`,
          creative_slot: `${creative_slot}-inner BGDM`,
        },
      ],
    };
    TriggerGaViewPromotion("select_promotion", datasLayer);
  };

  const handleClickRedirection = (isNewTab: boolean, path: URL | string) => {
    callSelectPromotionEvent();
    if (isNewTab) {
      window?.open(path);
    } else {
      router.push(path);
    }
  };
  return (
    <StyledGrid onClick={() => handleClickRedirection(isNewtab, path)}>
      <Styledimage alt={text} src={`${ReplaceImage(imageUrl)}`} />
      <BoxStyled>
        <TitleTypography $textColor={textColor}>{mobileTitle}</TitleTypography>
        <SubTitle ismobile={true}>{text}</SubTitle>
      </BoxStyled>
    </StyledGrid>
  );
};

const BuyingGuides = ({
  items,
  viewMore,
  title,
  bgColor,
  bgPadding,
  viewMoreLink,
  position,
  id,
  __component,
  promotion_id,
  promotion_name,
}: BuyingGuideInterface) => {
  const isMobile = useMobileCheck();
  const router = useRouter();
  const [hasFiredEvent, setHasFiredEvent] = useState(false);
  const viewEventWrapper = useRef();

  const { ref, inView } = useInView({
    triggerOnce: true,
    threshold: 0.25,
  });
  const [mobileCarouselRef, mobileInView] = useInView({
    triggerOnce: true,
    threshold: 0.25,
  });

  const cookie = new Cookies();
  const { getItem } = useStorage();

  useEffect(() => {
    if (inView && !hasFiredEvent) {
      const datasLayer = {
        previous_screen_name: "na",
        visitor_type: cookie.get("accessToken") ? "loggedin" : "Guest",
        user_id: getItem("customer_ref", "local") || "na",
        creative_name: `${items
          ?.map((item: any) => `${item?.creative_name}-outer BGDW`)
          .join("|")}`,
        creative_slot: `${items
          ?.map((item: any) => `${item?.creative_slot}-outer BGDW`)
          .join("|")}`,
        items: items?.map((item: any) => ({
          promotion_id: `${promotion_id}-inner BGDW`,
          promotion_name: `${promotion_name}-inner BGDW`,
          creative_name: `${item?.creative_name}-inner BGDW`,
          creative_slot: `${item?.creative_slot}-inner BGDW`,
        })),
      };
      TriggerGaViewPromotion("view_promotion", datasLayer);
      setHasFiredEvent(true);
    }
  }, [inView, hasFiredEvent, cookie, getItem, id, __component, items]);

  const initialSlides = isMobile ? 2 : 4;
  let newVisibleProducts: any = [];

  const trackGAEvent = (index: number) => {
    const roundedIndex = Math.round(index);
    const startIndex = roundedIndex;
    const endIndex = Math.min(roundedIndex + initialSlides, items.length);

    const visibleProducts = items.slice(startIndex, endIndex);

    const newItems = visibleProducts.filter(
      (item) => !newVisibleProducts.includes(item)
    );

    newVisibleProducts = [...newVisibleProducts, ...newItems];

    const datasLayer = {
      previous_screen_name: "na",
      visitor_type: cookie.get("accessToken") ? "loggedin" : "Guest",
      user_id: getItem("customer_ref", "local") || "na",
      creative_name: `${items
        ?.map((item: any) => `${item?.creative_name}-outer BGDM`)
        .join("|")}`,
      creative_slot: `${items
        ?.map((item: any) => `${item?.creative_slot}-outer BGDM`)
        .join("|")}`,
      items: newItems?.map((item: any) => ({
        promotion_id: `${promotion_id}-inner BGDM`,
        promotion_name: `${promotion_name}-inner BGDM`,
        creative_name: `${item?.creative_name}-inner BGDM`,
        creative_slot: `${item?.creative_slot}-inner BGDM`,
      })),
    };
    TriggerGaViewPromotion("view_promotion", datasLayer);
  };

  const handleClickRedirection = (
    isNewTab: boolean,
    path: URL | string,
    index: number
  ) => {
    if (isNewTab) {
      window?.open(path);
    } else {
      router.push(path);
    }
  };
  const userSettings = {
    arrows: false,
    dots: false,
    infinite: false,
    speed: 500,
    initialSlide: 0,
    slidesToShow: 2.1,
    slidesToScroll: 1,
    swipeToSlide: true,
  };

  const dataLayer = {
    item_id: "na",
    widget_type: widget_type,
    component_id: id,
    item_type: "na",
    item_name: "na",
    widget_title: __component,
    widget_description: "na",
    widget_postion: position,
    no_of_items: items?.length,
    index: position,
    item_brand: "na",
    item_category: "na",
    item_category2: "na",
    item_category3: "na",
    event_type: event_type,
    view_items: viewArray(items, `${__component}_${position}_`),
  };
  ViewEvent(viewEventWrapper, dataLayer, "view");

  const callEvent = (imageUrl: string, index?: any, linktext?: string) => {
    triggerGAEvent(
      {
        item_name: items[0]?.Item_name,
        item_id: `${__component}_${position}_${index + 1}`,
        widget_type: widget_type,
        item_type: items[0]?.Item_type,
        widget_title: __component,
        widget_description: "na",
        widget_postion: position,
        link_url: `${global?.window?.location?.origin}${items[0]?.path}`,
        link_text: linktext,
        no_of_items: items?.length,
        index: index + 1,
        item_brand: "na",
        item_category: "na",
        item_category2: "na",
        item_category3: "na",
        item_image_link: imageUrl,
        item_orginal_price: 0,
        item_price: 0,
        item_rating: "na",
        event_type: event_type,
      },
      "click"
    );
  };

  const callViewall = () => {
    triggerGAEvent(
      {
        widget_type: widget_type,
        widget_title: __component,
        widget_description: "na",
        widget_postion: position,
        link_url: `${global?.window?.location?.origin}${viewMoreLink}`,
        link_text: viewMore,
        no_of_items: items?.length,
        item_brand: "na",
        item_category: "na",
        item_type: items[0]?.Item_type,
        content_type: "product",
      },
      "view_all"
    );
  };

  const callSelectPromotionEvent = (
    creative_name: string,
    creative_slot: string
  ) => {
    const datasLayer = {
      previous_screen_name: "na",
      visitor_type: cookie.get("accessToken") ? "loggedin" : "Guest",
      user_id: getItem("customer_ref", "local") || "na",
      creative_name: `${creative_name}-outer BGDW`,
      creative_slot: `${creative_slot}-outer BGDW`,
      items: [
        {
          promotion_id: `${promotion_id}-inner BGDW`,
          promotion_name: `${promotion_name}-inner BGDW`,
          creative_name: `${creative_name}-inner BGDW`,
          creative_slot: `${creative_slot}-inner BGDW`,
        },
      ],
    };
    TriggerGaViewPromotion("select_promotion", datasLayer);
  };

  return (
    <MainBox
      p={isMobile ? "0 16px" : bgPadding}
      sx={{
        paddingTop: isMobile ? "12px" : "20px",
        paddingBottom: isMobile ? "12px" : "20px",
      }}
      bgcolor={bgColor}
      ref={viewEventWrapper}
    >
      {!isMobile && (
        <ViewAllbutton
          $backColor={bgColor}
          $isMobile={isMobile}
          onClick={() => {
            callViewall();
            return router.push(viewMoreLink);
          }}
        >
          {viewMore}
        </ViewAllbutton>
      )}
      {title && (
        <Box
          sx={{
            paddingTop: isMobile ? "2px" : "4px",
            paddingBottom: isMobile ? "4px" : "8px",
          }}
        >
          <Title interface={{ title: title, bgColor: bgColor }} />
        </Box>
      )}
      {!isMobile ? (
        <Box display="block" ref={ref}>
          <Stack
            spacing={{ lg: "18px", md: "14px", sm: "10px" }}
            direction="row"
          >
            {items?.map((item: ITEMS_LIST, index) => {
              const {
                imageUrl,
                text,
                path,
                isNewtab,
                textColor,
                creative_name,
                creative_slot,
              } = item;
              return (
                <StyledGrid
                  key={index}
                  onClick={() => {
                    callEvent(imageUrl, index, text);
                    callSelectPromotionEvent(creative_name, creative_slot);

                    handleClickRedirection(isNewtab, path, index);
                  }}
                  item
                  sm={12 / items?.length}
                  md={12 / items?.length}
                  lg={12 / items?.length}
                >
                  <Styledimage alt={text} src={`${ReplaceImage(imageUrl)}`} />
                  <SubTitleTypography
                    variant="subtitle1"
                    $textColor={textColor}
                    sx={{
                      fontSize: { lg: 16, md: 14, sm: 12 },
                      lineHeight: { lg: "24px", md: "20px", sm: "16px" },
                    }}
                  >
                    {text}
                  </SubTitleTypography>
                </StyledGrid>
              );
            })}
          </Stack>
        </Box>
      ) : (
        <div ref={mobileCarouselRef}>
          {mobileInView ? (
            <Carousel
              promotion_name={promotion_name}
              promotion_id={promotion_id}
              Component={MsiteCarousel}
              items={items}
              settings={userSettings}
              callBack={undefined}
              isVertical={undefined}
              trackGAEvent={trackGAEvent}
            />
          ) : null}
        </div>
      )}
      {isMobile && viewMore && (
        <StyledBox>
          <ViewAllbutton
            $backColor={bgColor}
            $isMobile={isMobile}
            onClick={() => {
              callViewall();
              return router.push(viewMoreLink);
            }}
          >
            {viewMore}
          </ViewAllbutton>
        </StyledBox>
      )}
    </MainBox>
  );
};

export default BuyingGuides;