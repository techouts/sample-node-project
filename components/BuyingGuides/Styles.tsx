import styled from "@emotion/styled";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import Button from "@mui/material/Button";
import Grid  from "@mui/material/Grid";
import { ColorType } from "../../utility/ColorType";
import { transientProps } from "../../utility/TransientProps";
export interface Responsive {
  ismobile: boolean;
}
export const MainBox = styled(Box)((theme) => ({
  width: "100%",
  "@media(max-width:600px)": {
    padding: "0  0 0 16px  ",
  },
}));
export const ViewAllbutton = styled(
  Button,
  transientProps
)<{ $backColor: string; $isMobile: boolean }>(({ $backColor, $isMobile }) => ({
  color:
    $backColor === "#231F20" ? "#231F20" : $isMobile ? "#DEA3B7" : "#AD184C",
  "@media(min-width:600px)": {
    position: "absolute",
    right: "5%",
    "&.MuiButtonBase-root.MuiButton-root": {
      paddingRight: "0px",
    },
  },
  "@media(max-width:600px)": {
    backgroundColor: "#231F20",
    padding: "6px 18px",
    display: "flex",
    top: "12px",
    marginBottom: "3%",
    "&:hover": {
      backgroundColor: "#231F20 !important",
      color: "#DEA3B7 !important",
    },
  },
  "&:hover": {
    backgroundColor: "#ffff",
    color: "#AD184C",
  },
  fontSize: "12px",
  borderRadius: "0px",
  letterSpacing: "1px",
  cursor: "pointer",
  float: "right",
}));
export const StyledGrid = styled(Grid)(() => ({
  position: "relative",
  paddingTop: "12px",
  width: "100%",
  "@media(max-width:600px)": {
    maxWidth: "100%",
    paddingTop: "10px",
    paddingLeft: "0px",
    paddingRight: "8px",
  },
}));
export const Styledimage = styled("img")(() => ({
  width: "100%",
  height: "100%",
  cursor: "pointer",
  borderRadius: "0px",
}));
export const StyledBox = styled(Box)(() => ({
  display: "flex",
  justifyContent: "center",
}));
export const BoxStyled = styled(Box)(() => ({
  width: "90%",
  margin: "auto",
  font: " -webkit-control",
}));
export const TitleTypography = styled(
  Grid,
  transientProps
)<{ $textColor: ColorType | string }>(({ $textColor }) => ({
  "@media(max-width:400px)": {
    fontSize: "12px",
  },
  position: "absolute",
  fontSize: "14px",
  fontWeight: 600,
  color: $textColor,
  bottom: "26px",
}));
export const SubTitleTypography = styled(
  Typography,
  transientProps
)<{ $textColor: ColorType | string }>(({ $textColor }) => ({
  "@media(max-width:1210px)": {
    padding: "0% 4% 4% 4%",
  },
  color: $textColor,
  fontSize: "16px",
  position: "absolute",
  bottom: 0,
  fontWeight: 400,
  padding: "0 12% 4.5% 14%",
  textAlign: "center",
}));
export const SubTitle = styled.p<Responsive>`
  font-size: 10px;
  font-weight: 600;
  overflow: hidden;
  display: inline-block;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 86%;
  margin: 1px;
  position: absolute;
  color: #e0e0e0;
  bottom: 10px;
`;
