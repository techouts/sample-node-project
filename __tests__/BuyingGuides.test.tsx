import { render } from "@testing-library/react";
import BuyingGuides from "../components/BuyingGuides/BuyingGuides";
import { ColorType } from "../utility/ColorType";
const color: ColorType = "#0000";

const items = [
  {
    id: 1,
    imageUrl:
      "https://ssbeauty-cms.s3.ap-south-1.amazonaws.com/Mask_Group_541093de18.png",
    text: "The art of applying eye shadow and choosing the correct color palette",
    textColor: color,
    subText: "",
    subTextColor: color,
    offerBackground: color,
    path: "/category/makeup",
    isNewTab: false,
    videoUrl: "",
    mobileTitle: "Palette and eyeshdow",
    Item_name: "Promotion",
    Item_type: "promotion",
    offerText: "",
    smallText: "",
    isNewtab: false,
  },
];

const buyingGuidesData = {
  id: 1,
  __component: "widget.grid-carousel",
  bgColor: "#FFFFFF",
  bgPadding: "0 5% 3%",
  title: "BUYING GUIDES",
  titleColor: color,
  showTextOnHover: "",
  showGradient: "",
  itemBackgroundSpacing: "",
  itemPaddingBottom: "",
  itemPaddingX: "",
  desktopColumns: 3,
  mobileColumns: 3,
  viewMore: "VIEW ALL",
  viewMoreLink: "/category/makeup",
  topViewMore: "",
  isBgColor: false,
  marginLeft: "",
  display: "",
  carouselDots: "",
  mobileItems: items,
  cardBorder: false,

  items: items,
  position: 21,
};
describe("Buying Guide", () => {
  it("renders Buying Guide static page and creates a snapshot of the page", () => {
    const { container } = render(<BuyingGuides {...buyingGuidesData} />);

    expect(container).toMatchSnapshot();
  });
});
