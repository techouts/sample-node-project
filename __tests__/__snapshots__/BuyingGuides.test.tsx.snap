// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us & Miscs Static Pages renders Buying Guide static page and creates a snapshot of the page 1`] = `
<div>
  <div
    class="MuiBox-root css-cpsh67"
  >
    <div
      class="MuiBox-root css-8atqhb"
    >
      <div
        class="MuiBox-root css-t7vv4g"
      >
        <div
          class="slick-slider slick-initialized"
          dir="ltr"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="MuiGrid-root css-1n3ut51-MuiGrid-root"
                  >
                    <img
                      alt="The art of applying eye shadow and choosing the correct color palette"
                      class="css-17di2j3"
                      src="https://images-cms.sit.shopper-stop.in/Mask_Group_541093de18.png"
                    />
                    <div
                      class="MuiBox-root css-15tyxld"
                    >
                      <div
                        class="MuiGrid-root css-9nketq-MuiGrid-root"
                      >
                        Palette and eyeshdow
                      </div>
                      <p
                        class="css-5tb5xs"
                      >
                        The art of applying eye shadow and choosing the correct color palette
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="MuiGrid-root css-1n3ut51-MuiGrid-root"
                  >
                    <img
                      alt="The art of applying eye shadow and choosing the correct color palette"
                      class="css-17di2j3"
                      src="https://images-cms.sit.shopper-stop.in/Mask_Group_1_3a5ae50eb7.png"
                    />
                    <div
                      class="MuiBox-root css-15tyxld"
                    >
                      <div
                        class="MuiGrid-root css-9nketq-MuiGrid-root"
                      >
                        Palette and eyeshdow
                      </div>
                      <p
                        class="css-5tb5xs"
                      >
                        The art of applying eye shadow and choosing the correct color palette
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="2"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="MuiGrid-root css-1n3ut51-MuiGrid-root"
                  >
                    <img
                      alt="The art of applying eye shadow and choosing the correct color palette"
                      class="css-17di2j3"
                      src="https://images-cms.sit.shopper-stop.in/buyingguidevideo_e38590ddcc.png"
                    />
                    <div
                      class="MuiBox-root css-15tyxld"
                    >
                      <div
                        class="MuiGrid-root css-9nketq-MuiGrid-root"
                      >
                        Palette and eyeshdow
                      </div>
                      <p
                        class="css-5tb5xs"
                      >
                        The art of applying eye shadow and choosing the correct color palette
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Buying Guide renders Buying Guide static page and creates a snapshot of the page 1`] = `
<div>
  <div
    class="MuiBox-root css-1ydgutl"
  >
    <div
      class="MuiBox-root css-14q5kwc"
    >
      <div
        class="MuiBox-root css-5p2tng"
      >
        BUYING GUIDES
      </div>
    </div>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <div
        class="MuiBox-root css-t7vv4g"
      >
        <div
          class="slick-slider slick-initialized"
          dir="ltr"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="MuiGrid-root css-1n3ut51-MuiGrid-root"
                  >
                    <img
                      alt="The art of applying eye shadow and choosing the correct color palette"
                      class="css-17di2j3"
                      src="https://images-cms.sit.shopper-stop.in/Mask_Group_541093de18.png"
                    />
                    <div
                      class="MuiBox-root css-15tyxld"
                    >
                      <div
                        class="MuiGrid-root css-9nketq-MuiGrid-root"
                      >
                        Palette and eyeshdow
                      </div>
                      <p
                        class="css-5tb5xs"
                      >
                        The art of applying eye shadow and choosing the correct color palette
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="MuiGrid-root css-1n3ut51-MuiGrid-root"
                  >
                    <img
                      alt="The art of applying eye shadow and choosing the correct color palette"
                      class="css-17di2j3"
                      src="https://images-cms.sit.shopper-stop.in/Mask_Group_1_3a5ae50eb7.png"
                    />
                    <div
                      class="MuiBox-root css-15tyxld"
                    >
                      <div
                        class="MuiGrid-root css-9nketq-MuiGrid-root"
                      >
                        Palette and eyeshdow
                      </div>
                      <p
                        class="css-5tb5xs"
                      >
                        The art of applying eye shadow and choosing the correct color palette
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="2"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="MuiGrid-root css-1n3ut51-MuiGrid-root"
                  >
                    <img
                      alt="The art of applying eye shadow and choosing the correct color palette"
                      class="css-17di2j3"
                      src="https://images-cms.sit.shopper-stop.in/buyingguidevideo_e38590ddcc.png"
                    />
                    <div
                      class="MuiBox-root css-15tyxld"
                    >
                      <div
                        class="MuiGrid-root css-9nketq-MuiGrid-root"
                      >
                        Palette and eyeshdow
                      </div>
                      <p
                        class="css-5tb5xs"
                      >
                        The art of applying eye shadow and choosing the correct color palette
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1wcjc1k"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-4gcdzy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        VIEW ALL
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
